qualification/conftest.py

```diff
@@ -159,17 +160,20 @@ def pytest_generate_tests(metafunc: pytest.Metafunc) -> None:
         metafunc.parametrize("band", metafunc.config.getini("bands"))
     if "n_channels" in metafunc.fixturenames or "narrowband_decimation" in metafunc.fixturenames:
         configs = [(int(n_channels), 1, False) for n_channels in metafunc.config.getini("wideband_channels")]
-        if not metafunc.definition.get_closest_marker("wideband_only"):
-            if metafunc.definition.get_closest_marker("no_vlbi"):
-                vlbi_params = [False]
-            else:
-                vlbi_params = [False, True]
-            configs.extend(
-                (int(n_channels), int(decimation), vlbi)
-                for decimation in metafunc.config.getini("narrowband_decimation")
-                for n_channels in metafunc.config.getini("narrowband_channels")
-                for vlbi in vlbi_params
-            )
+        configs.extend(
+            (int(n_channels), int(decimation), vlbi)
+            for decimation in metafunc.config.getini("narrowband_decimation")
+            for n_channels in metafunc.config.getini("narrowband_channels")
+            for vlbi in [False, True]
+        )
+        if metafunc.definition.get_closest_marker("wideband_only"):
+            configs = [config for config in configs if config[1] == 1]
+        if metafunc.definition.get_closest_marker("no_vlbi"):
+            configs = [config for config in configs if not config[2]]
+        if metafunc.definition.get_closest_marker("vlbi_only"):
+            configs = [config for config in configs if config[2]]
+        if not configs:
+            raise RuntimeError(f"Contradictory markers on test {metafunc.function.originalname}")
```

Comment:
Fair enough, it's condensed. I can't say I enjoyed reading it though 😅

The attached report only runs a single `vlbi_only` config. Have you tested to ensure the other parameter sets work fine?

Reply:
I think so. If I've screwed it up we'll find out soon enough.